Test/Tests/RealtimeClientChannelTests.swift

```diff
@@ -932,7 +934,7 @@ class RealtimeClientChannelTests: XCTestCase {
 
         waitUntil(timeout: testTimeout) { done in
             client.connection.once(.connected) { stateChange in
-                XCTAssertEqual(stateChange.reason?.code, ARTErrorCode.unableToRecoverConnectionExpired.intValue) // didn't resumed
+                XCTAssertNotEqual(oldConnectionId, client.connection.id) // didn't resumed
```

Comment:
>which would be an expected behaviour of the Realtime service

Did you mean "an unexpected"? Because otherwise it's in contradiction with the `RTN15c6` and `RTN15c7` you've mentioned. Realtime sends new connection id with no error:

<img width="659" alt="Screenshot 2023-08-19 at 23 47 32" src="https://github.com/ably/ably-cocoa/assets/45491869/afe9e281-93f6-467b-ac9a-4acdb5987c78">

Reply:
Sorry, I did mean "unexpected", yes. Are you saying that you've observed this unexpected behaviour? If so, please can you discuss with the Realtime team?

Reply:
Turned out it's bc of newly implemented [RTN8c/RTN9c](https://sdk.ably.com/builds/ably/specification/main/features/#RTN8c)

Reply:
Sorry, I’m afraid I don't understand your last comment at all.

Looking at the changes that this PR makes to `test__015__Channel__connection_state__should_attach_successfully_and_remain_attached_when_the_connection_state_without_a_successful_recovery_gets_CONNECTED`, I think that my original question still stands. This test is provoking a failed resume — why is there no error?

Reply:
I think both the purpose and the name of this test are slightly misleading. It claims it tests `RTL3d`, but instead addresses a bug in its behavior. I didn't find two other `RTL3d` tests doing a good job in testing this spec. The test aims to check if there is an endless loop for channel state transition after suspended (suspended > attached > suspended), which it does. It shouldn't check if there was a successful connection resume or not. So I've [removed](https://github.com/ably/ably-cocoa/pull/1714/commits/9ea873b79d82f3e8af8cb0b8499ed31cabbe35e0) this check and renamed the test. Also removed the `simulateLostConnectionAndState` call, because `id` and `key` are reset when the connection goes into suspended state (RTN8c/RTN9c).

Reply:
I've created [issue](https://github.com/ably/ably-cocoa/issues/1841) for this.